modules/json/format-json.c

```diff
@@ -51,19 +70,20 @@ tf_json_prepare(LogTemplateFunction *self, gpointer s, LogTemplate *parent,
   GOptionEntry format_json_options[] =
   {
     { "leave-initial-dot", 0, G_OPTION_FLAG_REVERSE, G_OPTION_ARG_NONE, &transform_initial_dot, NULL, NULL },
+    { "key-delimiter", 0, 0, G_OPTION_ARG_CALLBACK, _parse_key_delimiter, NULL, NULL },
```

Comment:
Can you add some unit tests please?

E.g.:
```c
Test(format_json, test_format_json_with_key_delimiter)
{
  assert_template_format("$(format-json --key-delimiter \"\t\" \".foo\t.b.a.r.\"=\"baz\")",
                         "{\".foo\":{\".b.a.r.\":\"baz\"}}");
  assert_template_format("$(format-json --key-delimiter \".\" \".foo.bar\"=\"baz\")",
                         "{\".foo\":{\"bar\":\"baz\"}}");
  assert_template_format("$(format-json \".foo.bar\"=\"baz\")",
                         "{\"_foo\":{\"bar\":\"baz\"}}");
  assert_template_format("$(format-json --key-delimiter=\"\" \".foo.bar\"=\"baz\")",
                         "{\"_foo\":{\"bar\":\"baz\"}}");
}

```

Reply:
I planned to you were just too fast with the review.  :)

Reply:
Added these as unit tests. Thanks for writing them.

I've changed some of these as I have changed --key-delimiter behaviour.